java/developer-guide/working-with-outlook-items/working-with-recurrences/_index.md

```diff
@@ -0,0 +1,527 @@
+~~~
+### **Weekly Recurrences: EndAfterDate Recurrance Type**
+"End By" option in the Mapi Task is achieved by setting the OccurrenceCount property calculated by the GetOccurrenceCount() function. This function takes start date , end date and RRULE string.
+#### **Weekly Recurrences: Setting the Every Day value**
+The following code snippet shows you how to set the period value to 1 and INTERVAL value to 1 in the RRULE string as well.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the weekly recurrence
+MapiCalendarWeeklyRecurrencePattern rec = new MapiCalendarWeeklyRecurrencePattern();
+rec.setEndType(MapiCalendarRecurrenceEndType.EndAfterDate);
+rec.setPatternType(MapiCalendarRecurrencePatternType.Week);
+rec.setPeriod(1);
+rec.setWeekStartDay(DayOfWeek.Sunday);
+rec.setDayOfWeek(MapiCalendarDayOfWeek.Friday);
+rec.setEndDate(endByDate);
+rec.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=WEEKLY;BYDAY=FR;INTERVAL=1"));
+~~~
+
+
+
+Every Day value can be set to any appropriate value and multiple days can be selected as shown in the following example:
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+MapiCalendarWeeklyRecurrencePattern record = new MapiCalendarWeeklyRecurrencePattern();
+record.setEndType(MapiCalendarRecurrenceEndType.EndAfterDate);
+record.setPatternType(MapiCalendarRecurrencePatternType.Week);
+record.setPeriod(2);
+record.setWeekStartDay(DayOfWeek.Sunday);
+record.setEndDate(endByDate);
+record.setDayOfWeek(MapiCalendarDayOfWeek.Friday | MapiCalendarDayOfWeek.Monday);
+record.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=WEEKLY;BYDAY=FR,MO;INTERVAL=2"));
+~~~
+### **Weekly Recurrences: NeverEnd Recurrance Type**
+End type can be set by using MapiCalendarRecurrenceEndType.NeverEnd. Period or INTERVAL can be set to required value say 1 in the following example.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the weekly recurrence
+MapiCalendarWeeklyRecurrencePattern recurrence = new MapiCalendarWeeklyRecurrencePattern();
+recurrence.setEndType(MapiCalendarRecurrenceEndType.NeverEnd);
+recurrence.setPatternType(MapiCalendarRecurrencePatternType.Week);
+recurrence.setPeriod(1);
+recurrence.setWeekStartDay(DayOfWeek.Sunday);
+recurrence.setDayOfWeek(MapiCalendarDayOfWeek.Friday);
+recurrence.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=WEEKLY;BYDAY=FR"));
+~~~
+## **Working with Monthly Recurrences**
+Aspose.Email supports creation of monthly recurrences using MapiCalendarMonthlyRecurrencePattern. Three different Mapi calendar recurrence end types can be used including EndAfterNOccurrences, EndAfterDate and NeverEnd. This section demonstrates the creation of different monthly recurrence patterns.
```

Comment:
`Aspose.Email supports the creation of monthly recurrences using *MapiCalendarMonthlyRecurrencePattern*. Three different Mapi calendar recurrence end types can be used including *EndAfterNOccurrences*, *EndAfterDate* and *NeverEnd*. This section demonstrates the creation of different monthly recurrence patterns.`